modules/analysis-common/src/main/java/module-info.java

```diff
@@ -21,5 +22,7 @@
 
     opens org.elasticsearch.analysis.common to org.elasticsearch.painless.spi; // whitelist resource access
 
+    exports org.elasticsearch.analysis.common.synonyms.action to org.elasticsearch.server; // Needed for synonyms actions execution
```

Comment:
We need to export this for elasticsearch server... I think having a separate Plugin for the API management would be beneficial - otherwise we'll be exposing parts of the Analysis to other non-related pieces.

Reply:
I think this only exports this Action from to server, we don't export anything else, right? I've checked other plugins do this export as well.

I am open to explore having a separate plugin, but for now I am more inclined to have everything with common-analysis module for these reasons:
- Synonyms API management is a part of the analysis and is not useful by itself as a standalone plugin
- From the common-analysis module, we would need to call SynonymsManaagementAPI functions, and it would be nice if all these classes are within the same plugin.

What do you think, Carlos?

Reply:
Sounds good to me, we can always create a separate plugin if needed be. 👍